src/Gui/View3DSettings.cpp

```diff
@@ -94,20 +94,46 @@ void View3DSettings::applySettings()
     OnChange(*hGrp,"UseVBO");
     OnChange(*hGrp,"RenderCache");
     OnChange(*hGrp,"Orthographic");
-    OnChange(*hGrp,"EnableHeadlight");
-    OnChange(*hGrp,"HeadlightColor");
-    OnChange(*hGrp,"HeadlightDirection");
-    OnChange(*hGrp,"HeadlightIntensity");
-    OnChange(*hGrp,"EnableBacklight");
-    OnChange(*hGrp,"BacklightColor");
-    OnChange(*hGrp,"BacklightDirection");
-    OnChange(*hGrp,"BacklightIntensity");
-    OnChange(*hGrp,"EnableFillLight");
-    OnChange(*hGrp,"FillLightColor");
-    OnChange(*hGrp,"FillLightDirection");
-    OnChange(*hGrp,"FillLightIntensity");
-    OnChange(*hGrp,"AmbientLightColor");
-    OnChange(*hGrp,"AmbientLightIntensity");
+
+    auto lightSourcesGrp = hGrp->GetGroup("LightSources");
+    OnChange(*lightSourcesGrp,"EnableHeadlight");
+    OnChange(*lightSourcesGrp,"HeadlightColor");
+    OnChange(*lightSourcesGrp,"HeadlightDirection");
+    OnChange(*lightSourcesGrp,"HeadlightIntensity");
+    OnChange(*lightSourcesGrp,"EnableBacklight");
+    OnChange(*lightSourcesGrp,"BacklightColor");
+    OnChange(*lightSourcesGrp,"BacklightDirection");
+    OnChange(*lightSourcesGrp,"BacklightIntensity");
+    OnChange(*lightSourcesGrp,"EnableFillLight");
+    OnChange(*lightSourcesGrp,"FillLightColor");
+    OnChange(*lightSourcesGrp,"FillLightDirection");
+    OnChange(*lightSourcesGrp,"FillLightIntensity");
+    OnChange(*lightSourcesGrp,"AmbientLightColor");
+    OnChange(*lightSourcesGrp,"AmbientLightIntensity");
+
+    // Workaround
+    // Clear old settings that was used for a while in 1.1dev
+    // By clearing these settings, 1.0 will be able to run with same config file again
+    // For more info: https://github.com/FreeCAD/FreeCAD/issues/19880
+    // TODO: Remove when 1.1.0 is about to be released
+    if (hGrp->GetASCII("FillLightDirection")[0] != '\0') {
```

Comment:
nullOrEmpty?

Reply:
As it won't be null, I thought this would be enough.

But changing it to your suggestion will make it slightly easier to read.

Reply:
That is exactly my point, null check is not needed but intent is clearer